### PR TITLE
feat(wallet): Implement market tab v2 designs

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/31611

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
https://github.com/brave/brave-core/assets/48117473/9c179445-fa80-44f0-b931-a4540364d1a1

Screenshot 2023-07-13 at 12 45 28

### CODE REVIEW

File: components/brave_wallet_ui/components/asset-name-and-icon/style.ts [hunk 1]
```diff
@@ -27,14 +28,22 @@ export const NameAndSymbolWrapper = styled.div`
 `
 
 export const AssetName = styled.span`
-  font-family: Poppins;
+  color: ${leo.color.text.primary};
+  font-family: Manrope;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/asset-name-and-icon/style.ts:32]: Can we check with design and make sure this was intentional?

File: components/brave_wallet_ui/components/asset-name-and-icon/style.ts [hunk 2]
```diff
@@ -27,14 +28,22 @@ export const NameAndSymbolWrapper = styled.div`
...
-  letter-spacing: 0.01em;
-  color: ${p => p.theme.color.text01};
+  letter-spacing: 0.28px;
 `
 
-export const AssetSymbol = styled(AssetName)`
-  color: ${p => p.theme.color.text03};
+export const AssetSymbol = styled.span`
+  color: ${leo.color.legacy.text2};
+  font-family: Manrope;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/asset-name-and-icon/style.ts:42]: Same here

File: components/brave_wallet_ui/components/asset-price-change/style.ts [hunk 3]
```diff
@@ -4,50 +4,31 @@
...
-  font-family: Poppins;
-  font-size: 11px;
-  font-weight: 400;
-  letter-spacing: 0.01em;
-  color: ${p => p.theme.color.background01};
-  
-  @media (prefers-color-scheme: dark) {
-    color: ${p => p.theme.color.text};
-  }
+  font-family: Manrope;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/asset-price-change/style.ts:20]: Same here

File: components/brave_wallet_ui/components/asset-price-change/style.ts [hunk 4]
```diff
@@ -4,50 +4,31 @@
...
-
-export const ArrowDown = styled(ArrowBase)`
-  background-image: url(${ArrowDownIcon});
-`
+export const Arrow = styled(Icon)<{
+  isDown: boolean
+}>`
+  --leo-icon-size: 24px;
+  color: ${(p) => p.isDown ? leo.color.systemfeedback.successIcon : leo.color.systemfeedback.errorIcon}
+`
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/asset-price-change/style.ts:34]: Missing end of page line

File: components/brave_wallet_ui/components/desktop/assets-filter-dropdown/style.ts [hunk 5]
```diff
@@ -22,30 +23,34 @@ export const Button = styled(WalletButton)`
...
-  border: ${(p) => `1px solid ${p.theme.color.interactive08}`};
+  background-color: ${leo.color.container.highlight};
   padding: 6px 12px;
+  border: none;
   border-radius: 8px;
   cursor: pointer;
 `
 
-export const Dropdown = styled.ul`
+export const Dropdown = styled.ul`
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/assets-filter-dropdown/style.ts:40]: There is an extra space after `

File: components/brave_wallet_ui/components/shared/market-grid/market-grid.style.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+  cursor: pointer;
+`
+
+export const Cell = styled.div<{
+  hideOnPanel?: boolean
+  hideOnSmall?: boolean
+}>`
+  display: flex;
+  align-items: center;
+  font-family: Manrope;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/shared/market-grid/market-grid.style.ts:100]: Same here

File: components/brave_wallet_ui/components/shared/market-grid/market-grid.style.ts [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+      case 'right':
+        return 'flex-end'
+      case 'center':
+        return 'center'
+      default:
+        return 'center'
+    }
+  }};
+  width: 100%;
+  font-family: Manrope;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/shared/market-grid/market-grid.style.ts:150]: Same here

File: components/brave_wallet_ui/stories/wallet-components.tsx [hunk 8]
```diff
@@ -296,4 +299,30 @@ export const _Tabs = () => {
...
+        isBuySupported={() => true}
+        isDepositSupported={() => false} 
+        onClickBuy={() => {}}
+        onClickDeposit={() => {}}
+        onUpdateIframeHeight={() => {}}
+        fiatCurrency={'USD'}
+      />
+
+    </div>
+  )
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/stories/wallet-components.tsx:327]: Missing end of page line

Comment (muliswilliam) [reply components/brave_wallet_ui/stories/wallet-components.tsx:327]: Adde new line...highlight here is wrong 😄

**Review by Douglashdaniel — APPROVED** [review]
++

**Pull request merged**